### ISSUE TITLE
[Question] Is it possible to change Azure endpoint?

### ISSUE THREAD

**Issue by yershalom** [pulumi/pulumi-azure#589]
Hi,

Is it possible to change Azure API endpoint? For example, if I'm choosing public as environment so it will go to api.azure.net or something like that.. Can I change the endpoint?

Thanks.

**Comment by leezen** [comment]
https://www.pulumi.com/docs/intro/cloud-providers/azure/#configuration describes the set of configuration options for the provider, which includes setting the environment and location.

**Comment by yershalom** [comment]
@leezen Thanks for your answer, but in my use case the endpoint is not under non of these environments.
That's why I'm asking if there is an option to pass a custom endpoint maybe?
Thanks

**Comment by leezen** [comment]
Ah, sorry for misunderstanding. No, I don't believe there's a way to pass in a custom endpoint the way you're describing. Are you trying to use Azure Stack by chance?

**Comment by yershalom** [comment]
Yes, I'm using Azure stack, but I want to use for some internal environment that we have inside of microsoft :-)

**Comment by lukehoban** [comment]
FWIW - there's some similar discussions related to the upstream Terraform AzureRM provider at https://github.com/terraform-providers/terraform-provider-azurerm/issues/6816.  I suspect that some of the same workarounds mentioned there might be possible for Pulumi as well.
